### ISSUE TITLE
TypeError: body is not iterable with function returning function

### ISSUE THREAD

**Issue by abe33** [rollup/rollup#1032]
The issue is only raised for fat arrow functions without braces that returns a function

For instance, here the smallest code I could find to trigger that issue: 

``` js
const f = (a) => (b) => {}

function ff (a) { return f(a) }
```

While this version is processed properly: 

``` js
const f = (a) => { return (b) => {} }

function ff (a) { return f(a) }
```

It seems that it doesn't matter that the returned function is a fat arrow function with or without braces or a plain old `function () {}`.

Here's the stack trace when running rollup on such sample: 
<details>

```
TypeError: body[Symbol.iterator] is not a function
    at fnHasEffects (/<redacted>/rollup/src/ast/nodes/shared/callHasEffects.js:16:18)
    at callHasEffects (/<redacted>/rollup/src/ast/nodes/shared/callHasEffects.js:34:9)
    at CallExpression.hasEffects (/<redacted>/rollup/src/ast/nodes/CallExpression.js:29:10)
    at CallExpression.isUsedByBundle (/<redacted>/rollup/src/ast/nodes/CallExpression.js:38:15)
    at /<redacted>/rollup/src/Bundle.js:150:31
    at process._tickCallback (internal/process/next_tick.js:103:7)
    at Module.runMain (module.js:592:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

</details>

**Comment by kzc** [comment]
Possible fix:
```patch
--- a/src/ast/nodes/shared/callHasEffects.js
+++ b/src/ast/nodes/shared/callHasEffects.js
@@ -11,7 +11,7 @@ function fnHasEffects ( fn ) {
 
        // handle body-less arrow functions
        const scope = fn.body.scope || fn.scope;
-       const body = fn.body.body || [ fn.body ];
+       const body = fn.body.body && Array.isArray(fn.body.body) ? fn.body.body : [ fn.body ];
 
        for ( const node of body ) {
                if ( node.hasEffects( scope ) ) {
```

**Comment by PaulBGD** [comment]
@kzc can confirm that this fixes the issue, PR it please!

**Comment by kzc** [comment]
Fix: #1062
